src/utp_stream.cpp

```diff
@@ -1673,7 +1673,18 @@ bool utp_socket_impl::send_pkt(int const flags)
 	}
 
 	if (!m_stalled)
+	{
 		++p->num_transmissions;
+		// Only reset the timeout for the initial packet
+		if (m_bytes_in_flight == 0 && (p->size > p->header_size || (flags & pkt_fin)))
+		{
+			m_timeout = clock_type::now() + milliseconds(packet_timeout());
```

Comment:
```suggestion
			m_timeout = now + milliseconds(packet_timeout());
```

Reply:
I think it would be better to use the timestamp we've already asked for a bit further up